modules/codegen/src/smithy4s/codegen/internals/SmithyToIR.scala

```diff
@@ -683,6 +683,28 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
 
     }
 
+  private def imputeZeroValuesOnDefaultTraits(
+      shape: Shape
+  ): List[Trait] = shape.getAllTraits().asScala.values.toList.map {
+    case default: DefaultTrait if default.toNode == Node.nullNode =>
+      val tpe = shape.asMemberShape().asScala match {
+        case Some(memShape) => model.getShape(memShape.getTarget).get.getType
+        case None           => shape.getType
+      }
+      val newNode = tpe match {
+        case ShapeType.STRING  => Node.from("")
+        case ShapeType.MAP     => Node.objectNode()
```

Comment:
what about ShapeType.LIST to  Node.arrayNode() ?

Reply:
Good catch, I will add tests for all types and fix that